src/Umbraco.Core/Services/IRuntime.cs

```diff
@@ -1,22 +1,15 @@
-﻿using System;
+using Microsoft.Extensions.Hosting;
 
 namespace Umbraco.Core
 {
     /// <summary>
     /// Defines the Umbraco runtime.
     /// </summary>
-    public interface IRuntime
+    public interface IRuntime : IHostedService
```

Comment:
@rustybox This seems to make a lot of sense 👏 My only concern was knowing that IHostedService.StartAsync always fires before any endpoint routing happens and having a look at the source code, that actually happens quite early in startup so that's all good and the front-end is still rendering :)